### ISSUE TITLE
All Dhatu Entries

### ISSUE THREAD

**Issue by Shalu411** [sanskrit-lexicon/SKD#8]
Hariom, 
Greetings to all. 
Having worked with VCP dhAtus lately, here is another request-
Now let's go for SKD.
    I need the whole list of all existent dhAtus, not just headwords but with inner explanatory text contents separately.. either each as a separate text file or whole as one file.. whichever is easy.
    Text should be in Devanagari inevitably.

    -Warm wishes-
    Shalu

**Comment by gasyoun** [comment]
> Text should be in Devanagari inevitably.

Is a matter of a single click.

**Comment by funderburkjim** [comment]
@Shalu411  Please look at the report [skd_verb1_deva](https://github.com/sanskrit-lexicon/SKD/blob/master/verbs01/skd_verb1_deva.txt).  (Note you can download or view 'raw').

[skd_verb1](https://github.com/sanskrit-lexicon/SKD/blob/master/verbs01/skd_verb1.txt) is the same but with slp1 spellings.

There are 2260 entries.  For each entry there is a header line and then the full text from skd.
Each of these is believed to be a root.
A sample header line is
`;; Case 2258: L=42186, k1=hlAda, k2=hlAda, vcp=hlAda, mw=hlAd`
* L the Cologne id
* k1 The root (key1) spelling, in slp1
* k2  The root (key2) spelling
* vcp The root spelling as in VCP.  This was found by seeing if the skd spelling matched
   a vcp root.   There were 217 non-matches, shown as `vcp=?`.   
   * conclusion:  in general, vcp and skd spell roots the same way.
* mw  The corresponding spelling in MW.  All but 20 of the 2260 are matched.  skd verbs with
  no matching show as `mw=?`.
  * of course, this matching is open to scrutiny and cprrrection.

## kavikalpadrumaH the key
Examination of several known roots suggested that skd roots are drawn from kavikalpadruma
with the phrase `iti kavikalpadrumaH` appearing in the first few lines of the text of records.
AFAIK,  this is the only filter needed to get the roots.   

There are only a very small number of cases where `iti kavikalpadrumaH` appears in the text
but the entry is not a root.  An example is `andUkaH`;  but this can be excluded as a verb because
it is marked as 'puM' (masculine).   Thus all cases marked as 'klI' (neuter), 'puM' (masculine),
'strI' (feminine), or 'tri'  [?? adjective]  were excluded.

## upasargas not found
I could find no definite instances of upasargas mentioned in the skd verb text, nor did I find
prefixed verbs as separate entries in skd.   This is surprising, so maybe prefixed verbs are 
mentioned in skd in some way that eluded my analysis.

**Comment by gasyoun** [comment]
So quick, so good.

> All but 20 of the 2260 are matched

So close.

> prefixed verbs are mentioned in skd in some way that eluded my analysis.

@drdhaval2785 any clue?

**Comment by drdhaval2785** [comment]
Kavikalpadruma did not handle upasargas. It seems that SKD has also not handlec upasargas.

The words formed by upasargas are handled in their own right rather than preverb+root.

**Comment by gasyoun** [comment]
> It seems that SKD has also not handlec upasargas.

Can it be?

> The words formed by upasargas are handled in their own right rather than preverb+root.

Sure, but they should still remain verbs, right? Can you think of an example?

**Comment by Shalu411** [comment]
Namaste Jim, 
It was great to see the file. Yes, SKD mostly had KKD.. We three had once worked on it some years ago- This file seems so much perfect initially! Still will look at it again deeper and will get back. Your analysis is very helpful. 
Would love to know if machine can give SKD dhatus, and VCP dhatus sid eby side (which ever are matching) with inner detail; For Eg. 
SKD- अंश¦, त् क विभाजने । अंशकरणे । इति कविकल्प-
<>द्रुमः । तालव्योपधः । विभाजन इति भाज त् क तु
<>पृथक्कृत्यां इत्यस्य रूपं । अंशयति अंशापयति
<>धनं वणिक् । इति दुर्गादासकृतधातुदीपिका ॥

VCP- अंश¦ विभाजने अद० चु० उभ० । अंशयति ते आंशिशत् त ।
<>अङ्कापयतीतिवत् आपुकि अंशापयतीत्येके । अच् अंशः
<>उ--अंशुः णिनि अंशी क्त अंशितः ।

..say- in two adjacent cells of an excel sheet-
And we can keep non-matching ones as separate entries.. 

That was my idea initially to compare all dhatus and prepare an excel sheet with inner detail.. 
VCP and SKD are the two major Skt-Skt-dictionaries. We can see others later- But atleast these two would be helpful. I am not aware of possibility.. But you could give all vcp and mw comparisons,  unasked; so am asking this to know, if that way we can make by machine?
-Shalu

**Comment by gasyoun** [comment]
> if that way we can make by machine?

Guess yes, because Jim has compared a lot in the past in regards of dhatus.

**Comment by Shalu411** [comment]
Hariom
A perfect file, as I told you before. Now it's checked thoroughly (as much as my limit permits). 
I didn't find any non-dhatu. Thanks for this one-go-complete-work..!! Jim, you are a magician. A thousand thanks for this. 
Just three issues of typo-errors, for which I will come back again. 
SKD has no upasargas, but it is useful for one of my dream-projects (personal) to compare dhatu-pathas- and make a good exhaustive dhatu-database.  
Apte Skt-Eng-Dic. has upasargas again. Would write a request for  that- but please do when your time and convenience permit.
Thanks a again. (1001..th time)
-Shalu

**Comment by Andhrabharati** [comment]
This issue's title attracted my notice!

Seen that the main points (in my view) talked about here are-
1. Extract the dhAtus form SKD text, with some 'clue'
2. Checking the extracted list to find non-dhAtus

Other sub-points (again in my view) are-
1. Comparing SKD list with MW list
2. Comparing SKD list with VCP list

@gasyun
is this issue concluded by all parties involved earlier here with Shalu's post, or is there some update on this somewhere else?
If there is no pending action by anyone (4 persons) involved earlier, I would like to say something.
Otherwise I shall have a look at other places [if any update link(s) suggested] before my posting.

**Comment by drdhaval2785** [comment]
I dont think anyone is working on anything here. You may proceed. Shalu is not returning to this work anytime soon.

**Comment by Andhrabharati** [comment]
ok, good to hear the point.

Just wanted to say that I had spent a little time today in cross-checking the main points I listed, and got more dhAtus from SKD text (@funderburkjim was almost there at the clue but missed it by an inch); Usha missed identifying a non-dhAtu in Jim's list.

Just posting my comparision here, for @funderburkjim to get the correct clue required to get the dhAtus.
[SKD dhAtus from first line.pdf](https://github.com/sanskrit-lexicon/SKD/files/7121958/SKD.dhAtus.from.first.line.pdf)

With due regard to his ability and analysing skills, I should not teach Jim anything, but just giving a hint through my file.
[I guess, the file speaks for itself.]

**Comment by Andhrabharati** [comment]
If someone wants to work on it, I can give the .txt file of this; otherwise for studying .pdf is enough.

**Comment by drdhaval2785** [comment]
txt is needed for machines. PDF is just for human consumption.

**Comment by Andhrabharati** [comment]
understood; that's why I said, "If someone wants to **work** on it".

**Comment by drdhaval2785** [comment]
We should be guided by the immortal words of Bhavabhūti :
ये नाम केचिदिह नः प्रथयन्त्यवज्ञां 
जानन्ति ते किमपि तान्प्रति नैष यत्नः ।
उत्पत्स्यते तु मम कोऽपि समानधर्मा 
कालो ह्ययं निरवधिर्विपुला च पृथ्वी ॥

There will be someone somewhere who would appreciate your work. 
It may be handled now, may be a century later. 
Our job is to put whatever we have done till now in open domain, so that someone somewhere sometime may carry that work forward.

So, even if someone is not going to WORK on it immediately, I would request you to put the working files here, so that it may be tackled.

**Comment by Andhrabharati** [comment]
I am touched by your citation.

Incidentally, it is interpreted that Bhavabhuti had mentioned it in anguish; same thing that I feel myself many a times!

**Comment by drdhaval2785** [comment]
Anguish is a strong emotion. Very few people channelize anguish and do something to reduce anguish of others in future. 
Good to see a helping hand in you @Andhrabharati .

**Comment by funderburkjim** [comment]
> I can give the .txt file of this

Yes please.  Looked at the pdf,  and anticipate the data will be useful whenever time permits work on this to resume.
However, the txt form will be much more useful to work with when that time comes.

So, request you to upload the text form now.

**Comment by Andhrabharati** [comment]
Glad to hear from you @funderburkjim, on this.
[SKD dhAtus from first line.txt](https://github.com/sanskrit-lexicon/SKD/files/7150060/SKD.dhAtus.from.first.line.txt)
[This is a tab separated file.]

Pl. note that I had made some corrections in my file and hence some spelling differences etc. between my text and your text.

**Comment by Andhrabharati** [comment]
You just need to take the 40 lines that are without corresponding text in both AB and Jim versions, and add to your **skd_verb1_deva** file.

And the 0133 in Jim's version is a non-dhAtu entry.

**Comment by funderburkjim** [comment]
Thanks for file and hint on the 40 missed.

**Comment by Andhrabharati** [comment]
Now, I [found](https://github.com/sanskrit-lexicon/SKD/issues/13#issuecomment-1800862024) 3 addl. dhatus, that were merged into the previous entries in the cdsl text. **And one of them is 'merged' in the SKD print itself!!**
